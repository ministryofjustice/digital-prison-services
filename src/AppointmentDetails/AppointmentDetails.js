import React from 'react'
import PropTypes from 'prop-types'
import { Container, Info, Text } from './AppointmentDetails.styles'
import { DayOfTheWeek, DayMonthYear, Time } from '../dateHelpers'
import { properCaseName } from '../utils'
import RecurringAppointments from '../BulkAppointments/RecurringAppointments'

const AppointmentDetails = ({
  appointmentTypeDescription,
  locationDescription,
  startTime,
  endTime,
  comments,
  recurring,
  repeats,
  times,
}) => (
  <Container>
    <Info>
      <strong>Appointment type</strong>
      <Text>{appointmentTypeDescription}</Text>
    </Info>

    <Info>
      <strong>Appointment location</strong>
      <Text>{locationDescription}</Text>
    </Info>

    <Info>
      <strong>Date and time </strong>
      <Text>
        <div>{DayOfTheWeek(startTime)}</div>
        {`${DayMonthYear(startTime)} - ${Time(startTime)}`} {endTime && ` - ${Time(endTime)}`}
      </Text>
    </Info>

    {comments && (
      <Info>
        <strong>Comments</strong>
        <Text>{comments.length > 15 ? `${comments.substr(0, 15)}...` : comments}</Text>
      </Info>
    )}

    {recurring && (
      <Info>
        <strong>Recurring</strong>
        <Text>
          {properCaseName(repeats)}, {times} occurrences
        </Text>
        <Text>Ends {RecurringAppointments.recurringEndDate({ startTime, repeats, times })}</Text>
      </Info>
    )}
  </Container>
)

AppointmentDetails.propTypes = {
  appointmentTypeDescription: PropTypes.string,
  locationDescription: PropTypes.string,
  startTime: PropTypes.string.isRequired,
  endTime: PropTypes.string,
  comments: PropTypes.string,
  recurring: PropTypes.bool,
  repeats: PropTypes.string,
  times: PropTypes.string,
}

AppointmentDetails.defaultProps = {
  comments: '',
  endTime: '',
  recurring: false,
  repeats: '',
  times: '',
  appointmentTypeDescription: '',
  locationDescription: '',
}

export default AppointmentDetails
