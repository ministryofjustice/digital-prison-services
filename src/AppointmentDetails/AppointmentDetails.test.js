import React from 'react'
import { shallow } from 'enzyme'

import AppointmentDetails from './AppointmentDetails'

describe('Appointment details', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <AppointmentDetails
        appointmentTypeDescription="ACT"
        locationDescription="GYM"
        comments="test"
        startTime="2019-03-20T:21:0:00Z"
        repeats="DAILY"
        times="5"
        recurring
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should hide comment section when there is no comment to display', () => {
    const wrapper = shallow(
      <AppointmentDetails appointmentTypeDescription="ACT" locationDescription="GYM" startTime="2019-10-10T:21:0:00Z" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
