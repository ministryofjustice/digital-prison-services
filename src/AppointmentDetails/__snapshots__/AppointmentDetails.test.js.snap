// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment details should hide comment section when there is no comment to display 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppointmentDetails
    appointmentTypeDescription="ACT"
    comments=""
    endTime=""
    locationDescription="GYM"
    recurring={false}
    repeats=""
    startTime="2019-10-10T:21:0:00Z"
    times=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <strong>
            Appointment type
          </strong>
          <Styled(div)>
            ACT
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Appointment location
          </strong>
          <Styled(div)>
            GYM
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Date and time 
          </strong>
          <Styled(div)>
            <div>
              Thursday
            </div>
            10/10/2019 - 21:00
             
            
          </Styled(div)>
        </Styled(div)>,
        "",
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Appointment type
            </strong>,
            <Styled(div)>
              ACT
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Appointment type",
            },
            "ref": null,
            "rendered": "Appointment type",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "ACT",
            },
            "ref": null,
            "rendered": "ACT",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Appointment location
            </strong>,
            <Styled(div)>
              GYM
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Appointment location",
            },
            "ref": null,
            "rendered": "Appointment location",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "GYM",
            },
            "ref": null,
            "rendered": "GYM",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Date and time 
            </strong>,
            <Styled(div)>
              <div>
                Thursday
              </div>
              10/10/2019 - 21:00
               
              
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Date and time ",
            },
            "ref": null,
            "rendered": "Date and time ",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div>
                  Thursday
                </div>,
                "10/10/2019 - 21:00",
                " ",
                "",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Thursday",
                },
                "ref": null,
                "rendered": "Thursday",
                "type": "div",
              },
              "10/10/2019 - 21:00",
              " ",
              "",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "",
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <strong>
              Appointment type
            </strong>
            <Styled(div)>
              ACT
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Appointment location
            </strong>
            <Styled(div)>
              GYM
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Date and time 
            </strong>
            <Styled(div)>
              <div>
                Thursday
              </div>
              10/10/2019 - 21:00
               
              
            </Styled(div)>
          </Styled(div)>,
          "",
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Appointment type
              </strong>,
              <Styled(div)>
                ACT
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Appointment type",
              },
              "ref": null,
              "rendered": "Appointment type",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "ACT",
              },
              "ref": null,
              "rendered": "ACT",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Appointment location
              </strong>,
              <Styled(div)>
                GYM
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Appointment location",
              },
              "ref": null,
              "rendered": "Appointment location",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "GYM",
              },
              "ref": null,
              "rendered": "GYM",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Date and time 
              </strong>,
              <Styled(div)>
                <div>
                  Thursday
                </div>
                10/10/2019 - 21:00
                 
                
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date and time ",
              },
              "ref": null,
              "rendered": "Date and time ",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <div>
                    Thursday
                  </div>,
                  "10/10/2019 - 21:00",
                  " ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Thursday",
                  },
                  "ref": null,
                  "rendered": "Thursday",
                  "type": "div",
                },
                "10/10/2019 - 21:00",
                " ",
                "",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "",
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Appointment details should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppointmentDetails
    appointmentTypeDescription="ACT"
    comments="test"
    endTime=""
    locationDescription="GYM"
    recurring={true}
    repeats="DAILY"
    startTime="2019-03-20T:21:0:00Z"
    times="5"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <strong>
            Appointment type
          </strong>
          <Styled(div)>
            ACT
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Appointment location
          </strong>
          <Styled(div)>
            GYM
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Date and time 
          </strong>
          <Styled(div)>
            <div>
              Wednesday
            </div>
            20/03/2019 - 21:00
             
            
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Comments
          </strong>
          <Styled(div)>
            test
          </Styled(div)>
        </Styled(div)>,
        <Styled(div)>
          <strong>
            Recurring
          </strong>
          <Styled(div)>
            Daily
            , 
            5
             occurrences
          </Styled(div)>
          <Styled(div)>
            Ends 
            Sunday, March 24th 2019
          </Styled(div)>
        </Styled(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Appointment type
            </strong>,
            <Styled(div)>
              ACT
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Appointment type",
            },
            "ref": null,
            "rendered": "Appointment type",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "ACT",
            },
            "ref": null,
            "rendered": "ACT",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Appointment location
            </strong>,
            <Styled(div)>
              GYM
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Appointment location",
            },
            "ref": null,
            "rendered": "Appointment location",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "GYM",
            },
            "ref": null,
            "rendered": "GYM",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Date and time 
            </strong>,
            <Styled(div)>
              <div>
                Wednesday
              </div>
              20/03/2019 - 21:00
               
              
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Date and time ",
            },
            "ref": null,
            "rendered": "Date and time ",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div>
                  Wednesday
                </div>,
                "20/03/2019 - 21:00",
                " ",
                "",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wednesday",
                },
                "ref": null,
                "rendered": "Wednesday",
                "type": "div",
              },
              "20/03/2019 - 21:00",
              " ",
              "",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Comments
            </strong>,
            <Styled(div)>
              test
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Comments",
            },
            "ref": null,
            "rendered": "Comments",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              Recurring
            </strong>,
            <Styled(div)>
              Daily
              , 
              5
               occurrences
            </Styled(div)>,
            <Styled(div)>
              Ends 
              Sunday, March 24th 2019
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Recurring",
            },
            "ref": null,
            "rendered": "Recurring",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Daily",
                ", ",
                "5",
                " occurrences",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Daily",
              ", ",
              "5",
              " occurrences",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Ends ",
                "Sunday, March 24th 2019",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Ends ",
              "Sunday, March 24th 2019",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <strong>
              Appointment type
            </strong>
            <Styled(div)>
              ACT
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Appointment location
            </strong>
            <Styled(div)>
              GYM
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Date and time 
            </strong>
            <Styled(div)>
              <div>
                Wednesday
              </div>
              20/03/2019 - 21:00
               
              
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Comments
            </strong>
            <Styled(div)>
              test
            </Styled(div)>
          </Styled(div)>,
          <Styled(div)>
            <strong>
              Recurring
            </strong>
            <Styled(div)>
              Daily
              , 
              5
               occurrences
            </Styled(div)>
            <Styled(div)>
              Ends 
              Sunday, March 24th 2019
            </Styled(div)>
          </Styled(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Appointment type
              </strong>,
              <Styled(div)>
                ACT
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Appointment type",
              },
              "ref": null,
              "rendered": "Appointment type",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "ACT",
              },
              "ref": null,
              "rendered": "ACT",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Appointment location
              </strong>,
              <Styled(div)>
                GYM
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Appointment location",
              },
              "ref": null,
              "rendered": "Appointment location",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "GYM",
              },
              "ref": null,
              "rendered": "GYM",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Date and time 
              </strong>,
              <Styled(div)>
                <div>
                  Wednesday
                </div>
                20/03/2019 - 21:00
                 
                
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date and time ",
              },
              "ref": null,
              "rendered": "Date and time ",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <div>
                    Wednesday
                  </div>,
                  "20/03/2019 - 21:00",
                  " ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wednesday",
                  },
                  "ref": null,
                  "rendered": "Wednesday",
                  "type": "div",
                },
                "20/03/2019 - 21:00",
                " ",
                "",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Comments
              </strong>,
              <Styled(div)>
                test
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Comments",
              },
              "ref": null,
              "rendered": "Comments",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                Recurring
              </strong>,
              <Styled(div)>
                Daily
                , 
                5
                 occurrences
              </Styled(div)>,
              <Styled(div)>
                Ends 
                Sunday, March 24th 2019
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recurring",
              },
              "ref": null,
              "rendered": "Recurring",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Daily",
                  ", ",
                  "5",
                  " occurrences",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Daily",
                ", ",
                "5",
                " occurrences",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Ends ",
                  "Sunday, March 24th 2019",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Ends ",
                "Sunday, March 24th 2019",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
